.github/workflows/on-merge.yml

```diff
@@ -100,6 +100,20 @@ jobs:
           version="${version#v}-alpha.${short_sha}"
           pulumictl dispatch -c pulumi-cli-dev-version -r pulumi/docs dev_version="$version"
 
+  tag-pkg:
+    name: tag-pkg
+    needs: [info, ci, update-dev-version]
+    if: ${{ needs.info.outputs.is-changelog-update }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Tag and push pkg version
+        run: |
+          version=${{ needs.info.outputs.pkg-version }}
+          git tag "pkg/${version}"
+          git push origin "pkg/${version}"
```

Comment:
Will this actually be the commit that lands in `master`? I was under the impression we'd need a workflow for when a commit is pushed to master.

Reply:
I think so, but I'm not sure.  The way I understand the merge queue is that it creates a new branch, with the commit that would land on master, and once CI is finished, it does a fast forward merge, so we'd have the same commit SHA.  It wouldn't be much harder to introduce a new workflow that happens on push to master, happy to try to switch to that.

Reply:
Changed that now.  The additional benefit is that it's a new CI job, so if it fails the only thing we need to do is to push the tag manually and then fix it, but releases won't fail.